source/contracts/reporting/ReportingWindow.sol

```diff
@@ -117,16 +118,21 @@ contract ReportingWindow is DelegationTarget, Typed, Initializable, IReportingWi
         }
 
         if (_state == IMarket.ReportingState.FINALIZED) {
-            if (!_market.isValid()) {
-                invalidMarketCount++;
-            }
-            marketReports.record(numberOfReportsByMarket[_market]);
-            totalWinningReportingTokens = totalWinningReportingTokens.add(_market.getFinalWinningReportingToken().totalSupply());
+            updateFinalizedMarket(_market);
```

Comment:
This should only be called if the market was previously not finalized and is now finalized.  I _suspect_ that this is probably only called in such a scenario, but I would be a lot more comfortable if we had some strong guarantees in the form of `require(...)` statements that ensured this method can only ever be called once per market.  If it is called multiple times for a market, we'll over-count a few things, some of which could be significant.